### ISSUE TITLE
Proposal: introduce a required flag to trigger installation from a package index

### ISSUE THREAD

**Issue by ghost** [pypa/setuptools#235]
Originally reported by: **mitchellrj (Bitbucket: [mitchellrj](http://bitbucket.org/mitchellrj), GitHub: [mitchellrj](http://github.com/mitchellrj))**

---

Currently, the `easy_install` command accepts either a local path (absolute or relative) as an argument, or the name of a package to be retrieved from a package index, defaulting to PyPI.

I propose that a new flag be introduced to indicate that the argument should be interpreted as a package name to be retrieved from a package index.

Consider the situation where a user mis-types:

```
sudo easy_install src
```

where there is no `src` directory relative to the user's current working directory. This will trigger `easy_install` to fetch the package named `src` from PyPI and run its setup script with root privileges.

Forcing the user to instead type:

```
sudo easy_install -pi foo
```

to install a package from a package index and only allowing the previous usage to install packages from the local filesystem eliminates this potential security risk.

Making a similar change to require a flag for local installation introduces a similar risk that the user forgets to use the flag and the remote package is still downloaded and run. Introducing a flag for installation from a package index means that even if the user omits the flag, the worst case scenario is that they install something from their own filesystem.

I have raised [a similar issue](https://github.com/pypa/pip/issues/1940) for `pip`.

TL;DR: Package managers should never download & run remote code just because a user mistypes a filename.

---
- Bitbucket: https://bitbucket.org/pypa/setuptools/issue/235

**Comment by abravalheri** [comment]
Hello, I suppose we can close this issue now that `easy_install` has been deprecated and widely replace by other installation means, such as `pypa/pip`.

If anyone would like to reopen this issue, please feel free to write a new comment with more information or other use cases that we might be missing 😄.
